crates/sui-json/src/lib.rs

```diff
@@ -160,22 +160,57 @@ impl SuiJsonValue {
     }
 
     fn to_move_value(val: &JsonValue, ty: &MoveTypeLayout) -> Result<MoveValue, anyhow::Error> {
+        // Double check that all numbers conform
+        if let JsonValue::Number(n) = val {
+            if n.as_u64().is_none() {
+                return Err(anyhow!(format!(
+                    "{} is not a valid number. Only u64 allowed.",
+                    n
+                )));
+            }
+        }
+
         Ok(match (val, ty) {
             // Bool to Bool is simple
             (JsonValue::Bool(b), MoveTypeLayout::Bool) => MoveValue::Bool(*b),
 
             // In constructor, we have already checked that the JSON number is unsigned int of at most U64
-            // Hence it is okay to unwrap() numbers
-            (JsonValue::Number(n), MoveTypeLayout::U8) => {
-                MoveValue::U8(u8::try_from(n.as_u64().unwrap())?)
-            }
-            (JsonValue::Number(n), MoveTypeLayout::U16) => {
-                MoveValue::U16(u16::try_from(n.as_u64().unwrap())?)
-            }
-            (JsonValue::Number(n), MoveTypeLayout::U32) => {
-                MoveValue::U32(u32::try_from(n.as_u64().unwrap())?)
-            }
-            (JsonValue::Number(n), MoveTypeLayout::U64) => MoveValue::U64(n.as_u64().unwrap()),
+            (JsonValue::Number(n), MoveTypeLayout::U8) => match n.as_u64() {
+                Some(x) => MoveValue::U8(u8::try_from(x)?),
+                None => {
+                    return Err(anyhow!(format!(
+                        "{} is not a valid number. Only u64 allowed.",
+                        n
+                    )))
```

Comment:
we can do 
```suggestion
                    return Err(anyhow!(
                        "{} is not a valid number. Only u64 allowed.",
                        n
                    ))
```
or  even simpler

```suggestion
                    anyhow::bail!(
                        "{} is not a valid number. Only u64 allowed.",
                        n
                    )
```